actions-router.js

```diff
@@ -0,0 +1,97 @@
+const express = require('express');
+
+const router = express.Router();
+
+const Actions = require('./data/helpers/actionModel.js')
+const Projects = require('./data/helpers/projectModel.js')
+
+
+//GET actions
+
+router.get('/', (req, res) => {
+    Actions.get()
+    .then(actions => {
+        res.status(200).json(actions)
+    })
+    .catch(error => {
+        res.status(500).json({ message: "no actions"})
+    })
+})
+
+router.get('/:id', (req, res) => {
+    const id = req.params.id;
+    Projects.getProjectActions(id)
+        .then(action => {
+            res.status(200).json(action)
+        })
+        .catch(error => {
+            res.status(500).json({ message: "unable to retrieve action"})
+        })
+});
+
+//POST Action
+
+router.post('/:project_id', (req, res) => {
+    //newAction = req.body;
+    const {project_id} = req.params;
+    const objToPass = {
+        project_id,
+        description: req.body.description,
+        notes: req.body.notes,
+        completed: req.body.completed
+    }
+    console.log(objToPass)
+    Actions.insert(objToPass)
+    .then(action => {
+        res.status(201).json(action)
+    })
+    .catch(error => {
+        res.status(500).json({ message: "unable to add new action" })
+    })
+    }   
+);
+
+//PUT update action
+
+router.put('/:id', (req, res) => {
+    const id = req.params.id;
+    const changes = req.body;
+    Actions.update(id, changes)
+        .then(project =>{
+            res.status(200).json(project)
+        })
+        .catch(error => {
+            res.status(500).json({ message: "unable to add post"})
+        })
+});
+
+//DELETE action
+
+router.delete('/:id', (req, res) => {
+    const id = req.params.id;
+    Actions.remove(id)
+        .then(action => {
+            res.status(200).json({message: "Action deleted"})
+        })
+        .catch(error => {
+            res.status(500).json({message: "unable to delete action"})
+        })
+})
+
+function validateProjectExistence(req, res, next) {
```

Comment:
good use case for middleware! i know i asked you to take it out when we were debugging :/